src/test/java/streams/part1/exercise/Exercise2.java

```diff
@@ -0,0 +1,141 @@
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.junit.Assert.assertEquals;
+
+@SuppressWarnings({"ConstantConditions", "unused"})
+public class Exercise2 {
+
+    @Test
+    public void calcAverageAgeOfEmployees() {
+        List<Employee> employees = getEmployees();
+
+        Double expected = employees
+                .stream()
+                .map(Employee::getPerson)
+                .collect(Collectors.averagingInt(Person::getAge));
+
+        assertEquals(33.66, expected, 0.1);
+    }
+
+    @Test
+    public void findPersonWithLongestFullName() {
+        List<Employee> employees = getEmployees();
+
+        @Value
+        class PersonIntegerPair {
+            Person person;
+            Integer integer;
+        }
+
+        Person expected = employees
+                .stream()
+                .map(Employee::getPerson)
+                .map(person -> new PersonIntegerPair(person, person.getFullName().length()))
+                .max(Comparator.comparingInt(PersonIntegerPair::getInteger))
+                .map(PersonIntegerPair::getPerson)
+                .orElseThrow(RuntimeException::new);
+
+        assertEquals(expected, employees.get(1).getPerson());
+    }
+
+    @Test
+    public void findEmployeeWithMaximumDurationAtOnePosition() {
+        List<Employee> employees = getEmployees();
+
+        @Value
+        class PersonJobHistoryEntryPair {
+            Employee employee;
+            JobHistoryEntry jobHistoryEntry;
+        }
+
+        Employee expected = employees
+                .stream()
+                .flatMap(employee
+                        -> employee
+                        .getJobHistory()
+                        .stream()
+                        .map(jobHistoryEntry -> new PersonJobHistoryEntryPair(employee, jobHistoryEntry)))
+                .max(Comparator.comparingInt(value -> value.jobHistoryEntry.getDuration()))
+                .map(PersonJobHistoryEntryPair::getEmployee)
+                .orElseThrow(RuntimeException::new);
+
+        assertEquals(expected, employees.get(4));
+    }
+
+    /**
+     * Вычислить общую сумму заработной платы для сотрудников.
+     * Базовая ставка каждого сотрудника составляет 75_000.
+     * Если на текущей позиции (последняя в списке) он работает больше трех лет - ставка увеличивается на 20%
+     */
+    @Test
+    public void calcTotalSalaryWithCoefficientWorkExperience() {
+        List<Employee> employees = getEmployees();
+
+        Double expected = employees.stream()
+                .map(Employee::getJobHistory)
+                .reduce(0.0,
+                        (res, list)
+                                -> res +
+                                ((list.get(list.size() - 1).getDuration() > 3) ?
+                                        75000 * 1.2 :
+                                        75000),
+                        Double::sum);
```

Comment:
Oh. I forgot about this fragment. It's interesting solution, but also it's hard to read this.

Better to keep at one line logically coupled information (like lambda declaration: ```(res, list) ->```) and extract multilined expressions into methods or local variables with an obvious name.

Also it's a good style to use constants instead of magic numbers like "3", "75000", "1.2".